src/cmdstan/command_helper.hpp

```diff
@@ -354,7 +448,18 @@ void get_theta_hat_csv(const std::string &fname,
   boost::algorithm::split(values, line, boost::is_any_of(","),
                           boost::token_compress_on);
   in.close();
-
+  // validate
+  if (!is_optimization) {
+    msg << "CSV file is not output from Stan optimization" << std::endl;
+    throw std::invalid_argument(msg.str());
+  }
+  // optimization output has col for "lp__" + model variables
+  if (names.size() < cparam_names.size()
+      || values.size() < cparam_names.size()) {
```

Comment:
Is there a reason these are `<` and not `!=`? I guess they won't necessarily be the same length until after we've removed any of the `__` names, but it seems like it should also be a problem if they're too long.

I think if you move this block below where we calculate `col_offset`, we should instead check that `names.size() - col_offset == cparam_names.size()`, and that `values.size() == names.size()`.

This should fix the above segfault(s)

Reply:
the optimization output may include transformed parameters and generated quantities values as well.
it needs to be at least as big as the number of constrained parameters, that's all we know.

Reply:
Good point. I still think it is essential to take into account `col_offset` in these comparisons and to compare `values` and `names` (not `values` and `cparam_names`) to avoid the later indexing being unsafe